app/config/custom_tags.yml

```diff
@@ -0,0 +1,87 @@
+ezpublish:
+    system:
+        admin_group:
+            fieldtypes:
+                ezrichtext:
+                    custom_tags: [ezyoutube, eztwitter, ezfacebook]
+
+    ezrichtext:
+        # Define Custom Tags
+        custom_tags:
+            ezyoutube:
+                template: '@ezdesign/fields/ezrichtext/custom_tags/ezyoutube.html.twig'
```

Comment:
Is this template supposed to be used on frontend and / or backend?

Reply:
It's frontend, but backoffice when previewing (not editing) content uses frontend templates. This is a perfect example showing the power of eZ Design Engine. 

1. Create separate design for frontend and separate for backend with separate themes.
e.g.:
```yml
ezpublish:
    system:
        site_group:
            design: app
        admin_group:
            design: admin
design_list:
    app: [app, tastefulplanet]
    admin: [admin, tastefulplanet]
```

2. Then create the following templates for front and admin respectively:
- `app/Resources/views/themes/app/fields/ezrichtext/custom_tags/ezyoutube.html.twig`
- `app/Resources/views/themes/admin/fields/ezrichtext/custom_tags/ezyoutube.html.twig`

3. Remove `ezyoutube.html.twig` from `tastefulplanet` theme.
4. Clear Symfony cache.

You should get different template on frontend and different in Admin UI view.

Reply:
Indeed, it's using the same template for frontend and backend for `content / view`. I tested it and it works.
My initial question was for `content / edit`.

Reply:
> My initial question was for `content / edit`.

`content / edit` preview needs some work both for backend and frontend (js) part. AFAIR this was supposed to be a separate effort to be scheduled in the future. 
// cc @bdunogier

Reply:
Thanks @alongosz for clarifying this point